djangocms_references/helpers.py

```diff
@@ -257,7 +243,14 @@ def apply_additional_modifiers(queryset):
     return queryset
 
 
-def get_all_reference_objects(content, draft_and_published=False):
+def apply_filters(queryset, model, state_selected):
+    if get_versionable_for_content(model):
```

Comment:
```suggestion
    if get_versionable_for_content(queryset.model):
```
The model can be different per queryset, not all items will be versionable, by using the user supplied model we are assuming every queryset is of the same model that we are looking for references for. 

For example, I'm looking for all references of a Page, The model is then Page. The Page could be found inside Alias, Navigation, some other none versioned model etc. Each queryset the model would then be:  Alias, Navigation, some other model. Not Page which is what this is currently doing.